panel/applets/places-indicator/PlaceItem.vala

```diff
@@ -0,0 +1,29 @@
+/*
+ * This file is part of budgie-desktop
+ *
+ * Copyright (C) 2015-2016 Solus Project
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ */
+
+public class PlaceItem : ListItem
+{
+    public PlaceItem(GLib.File file, string class)
+    {
+        item_class = class;
+        item_location = file.get_uri();
+
+        // Get and set the appropriate icon
+        GLib.FileInfo? info = null;
+        try {
+            info = file.query_info("standard::symbolic-icon", GLib.FileQueryInfoFlags.NONE, null);
+        } catch (GLib.Error e) {
+            message(e.message);
+        }
+
+        set_button(file.get_basename(), get_icon(info.get_symbolic_icon()));
```

Comment:
Potential null-pointer de-reference, due to info being null and caught the error.